components/eventing-controller/pkg/backend/nats/jetstream/jetstream.go

```diff
@@ -177,13 +177,53 @@ func (js *JetStream) SyncSubscription(subscription *eventingv1alpha1.Subscriptio
 		return err
 	}
 
+	if err := js.checkSubscriptionConfig(subscription, log); err != nil {
+		return err
+	}
+
 	if err := js.checkNATSSubscriptionsCount(subscription); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// CheckSubscriptionConfig checks that the latest Subscription Config changes are propagated to the consumer.
+// In our case config contains only the "maxInFlightMessages" property, which is the maxAckPending on the consumer side.
+func (js *JetStream) checkSubscriptionConfig(subscription *eventingv1alpha1.Subscription, log *zap.SugaredLogger) error {
```

Comment:
I would really stick to the one I picked, since it doesn't update the consumer every time, but only in case the config changed, which happens not so often.